docs/content/interacting/relationship-queries.mdx

```diff
@@ -268,32 +269,60 @@ From there, we will learn that
 
 The Expand call is expensive and has high latency. As such, it is designed to be used for debugging and understanding why a user has a particular relationship with a specific object. It is not designed for checking whether a user has a particular relationship with a specific object. In that case the Check API call should be used instead.
 
+## ListObjects
+
+### What Is It For?
+
+The [ListObjects API](/api/service#/Relationship%20Queries/ListObjects) is an API endpoint that returns the list all of 
+the objects (of a particular type) that a user has a relationship with. 
+
+It provides a solution to the [Search with Permissions (Option 3)](./search-with-permissions.mdx#option-3-build-a-list-of-ids-then-search) use case for access-aware filtering on smaller object collections.
+
+### When To Use?
+
+Use the ListObjects API to filter the objects that a user can see based on the relationships they have.
+
+See [Search with Permissions](./search-with-permissions) for more guidance.
+
+<ListObjectsRequestViewer objectType="document" relation="reader" user="bob" contextualTuples={[{object: "document:otherdoc", relation: "reader", "user": "bob"}]} />
+
+### Caveats And When Not To Use It
+
+The ListObjects API should be used when the number of objects of a certain type the user could have access to is low (~ 1000), and the percentage of the total objects that the user can have access to is also low.
+
+See [Search with Permissions](./search-with-permissions) for more guidance.
+
```

Comment:
Can you add our implementation specific caveats?

If configured, it might not return all objects, not optimized and latency is high, etc..?

Reply:
I've updated this section to include a description of the behavior of those configurations and how they impact the server.

WDYT about the updated content?

Reply:
There's some weird spacing and line breaks here.

Also I would reword this, maybe:

> The current implementation of `ListObjects` is not optimized and will take take time to resolve.
>
> In case the server is configured with `listObjectsDeadline` or `listObjectsMaxResults`, the returned results may be incomplete, as the server will return whatever results it found as soon it hits the configured deadline, or finds the configured maximum number of results.

Reply:
re-worded